libraries/AP_TemperatureSensor/AP_TemperatureSensor.cpp

```diff
@@ -142,6 +142,9 @@ void AP_TemperatureSensor::init()
 
         // call init function for each backend
         if (drivers[instance] != nullptr) {
+            // param count could have changed
+            AP_Param::invalidate_count();
```

Comment:
Why not call this at the end rather than inside the loop?

Reply:
I could..

Reply:
changed to only run once